lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/dao/ChatDao.kt

```diff
@@ -0,0 +1,177 @@
+                    Message.messageTimestamp AS latestMessageTimestamp,
+                    op.personUid AS otherPersonUid,
+                    op.firstNames AS otherPersonFirstNames,
+                    op.lastName AS otherPersonLastName,
+                    (0) AS unreadMessageCount,
+        
+                 (SELECT COUNT(*)
+                  FROM ChatMember mm
+                  WHERE mm.chatMemberChatUid = Chat.chatUid ) AS numMembers
+             FROM ChatMember
+             LEFT JOIN Chat ON Chat.chatUid = ChatMember.chatMemberChatUid
+             LEFT JOIN Message ON Message.messageUid =
+                 (SELECT messageUid
+                  FROM Message
+                  WHERE messageEntityUid = Chat.chatUid
+                      AND messageTableId = ${Chat.TABLE_ID}
+                  ORDER BY messageTimestamp DESC
+                  LIMIT 1)
+             LEFT JOIN Person op ON op.personUid =
+                 (SELECT pp.personUid
+                  FROM ChatMember cm
+                  LEFT JOIN Person pp ON pp.personUid = cm.chatMemberPersonUid
+                  WHERE cm.chatMemberChatUid = Chat.chatUid
+                      AND cm.chatMemberPersonUid != :personUid
+                      AND cm.chatMemberLeftDate = ${Long.MAX_VALUE}
+                  LIMIT 1)
+             WHERE ChatMember.chatMemberPersonUid = :personUid
+                 AND ChatMember.chatMemberLeftDate = ${Long.MAX_VALUE}
+                 AND Chat.chatUid != 0 
+                 
+             ORDER BY Message.messageTimestamp DESC)
+        SELECT Chats.*
+        FROM Chats
+        UNION
+        SELECT *
+        FROM
+            (SELECT Chat.*,
+                    '' AS latestMessage,
+                    0 AS latestMessageTimestamp,
+                    Person.personUid AS otherPersonUid,
+                    Person.firstNames AS otherPersonFirstNames,
+                    Person.lastName AS otherPersonLastName,
+                    0 AS unreadMessageCount,
+                    0 AS numMembers
+             FROM Person
+             LEFT JOIN Chat ON Chat.chatUid = 0
+             WHERE :searchBit != '%'
+                 AND Person.personUid != :personUid
+                 AND Person.personUid NOT IN
+                     (SELECT p.personUid
+                      FROM ChatMember c
+                      LEFT JOIN Person p ON p.personUid = c.chatMemberPersonUid)
+                 AND Person.firstNames||' '||Person.lastName LIKE :searchBit )
+        ORDER BY latestMessageTimestamp DESC
+    """)
+    abstract fun findAllChatsForUser(searchBit: String, personUid: Long)
+        : DoorDataSourceFactory<Int, ChatWithLatestMessageAndCount>
+
+
+    @Query("""
+        SELECT CASE
+                   WHEN Chat.isChatGroup THEN Chat.chatTitle
+                   ELSE Person.firstNames||' '||Person.lastName
+               END AS title
+        FROM Chat
+        LEFT JOIN Person ON Person.personUid =
+          (SELECT pp.personUid
```

Comment:
The person wouldn't need to be joined if this is a group. E.g.
LEFT JOIN Person 
          ON CAST(Chat.isChatGroup AS INTEGER) =1 
          AND Person.personUid = ...